migrations/2024-11-10-134311_smoosh-tables-together/up.sql

```diff
@@ -1,108 +1,33 @@
--- For each new actions table, transform the table previously used for the most common action type
--- into the new actions table, which should only change the table's metadata instead of rewriting the
--- rows
-ALTER TABLE comment_like RENAME TO comment_actions;
```

Comment:
This table rename technique should be kept. It avoids rewrites for a lot of rows. It also introduces the overhead of validating the new constraints for all rows, but that can be fixed by adding `NOT VALID` to the end of each new constraint (e.g. `ADD CONSTRAINT post_actions_check_liked CHECK ((liked IS NULL) = (like_score IS NULL)) NOT VALID`). To let PostgreSQL assume the constraints are true, they can be turned to normal constraints after migrations using `VALIDATE CONSTRAINT` (e.g. `ALTER TABLE post_actions VALIDATE CONSTRAINT post_actions_check_liked`).

Reply:
That takes about 4 days of downtime, so its not doable. The current strategy is to create empty new tables, retain the old tables, only copy the last month of data into the new ones, then have a background job copy in batches from the old to the new.